### PR TITLE
Fix offwaketime PID / TGID handling

### PR DESCRIPTION
**Pull request by palmtenor**
Currently, `offwaketime.py` uses the same TGID for symbol resolution. But I think the waker and target theoretically could belong to different TGID and we would get wrong symbols. I wonder if this would help for #1679 (@joelagnel can you try?), but it's the right thing to do anyways.

Also improve documentation a bit.

### CODE REVIEW

**Comment by yonghong-song** [comment]
cc @joelagnel

**Comment by palmtenor** [comment]
I will apply same change of #1692 after this lands. That might also cause incorrect stacks being displayed

**Comment by joelagnel** [comment]
Hi,  Sorry for the delays in replying. I am going through conference
traveling right now, will reply ASAP once I am back (which is 1-2 days).

**Comment by palmtenor** [comment]
@joelagnel No worries. Just want to let you know. We should fix this regardless of if it's root cause of your issue or not:))

File: tools/offwaketime.py [hunk 1]
```diff
@@ -114,58 +114,73 @@ def signal_ignore(signal, frame):
...
+
     wokeby.update(&pid, &woke);
     return 0;
 }
 
 int oncpu(struct pt_regs *ctx, struct task_struct *p) {
-    u32 pid = p->pid, t_pid=pid;
+    // PID and TGID of the previous Process (Process going into waiting)
+    u32 pid = p->pid;
     u32 tgid = p->tgid;
```
Comment (yonghong-song) [inline tools/offwaketime.py]: looks like we do not need tgid assignment here.

Comment (palmtenor) [reply tools/offwaketime.py]: We don’t use it directly here, but might be used in THREAD_FILTER

**Pull request merged**